Iterable/iterable-web-sdk
# [MOB-9520]: enable tree-shaking

## JIRA Ticket(s) if any

* [MOB-9520](https://iterable.atlassian.net/browse/MOB-9520)

## Description

Enables tree shaking to resolve this [issue](https://github.com/Iterable/iterable-web-sdk/issues/442) 

## Test Steps
1. Remove `sideEffects: false` from `package.json`. 
2. Run `yarn webpack-analyze` from `react-example` dir
3. Note the bundle size
4. Add back `sideEffects:false` to `package.json`
5. See smaller bundle size due to tree shaking

Before: 3.31 mb
<img width="1673" alt="Screenshot 2024-09-11 at 10 33 04 AM" src="https://github.com/user-attachments/assets/7faa8009-a047-4c22-b81b-4534b8997239">

After: 2.78 mb
<img width="1679" alt="Screenshot 2024-09-11 at 10 32 45 AM" src="https://github.com/user-attachments/assets/c3b54688-d5d6-42d8-ac46-85749bfef435">


[MOB-9520]: https://iterable.atlassian.net/browse/MOB-9520?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
